agent/service.go

```diff
@@ -84,6 +101,58 @@ func (ts *TaskService) Create(ctx context.Context, req *shimapi.CreateTaskReques
 	return resp, nil
 }
 
+func (ts *TaskService) proxyStdio(ctx context.Context, stdin, stdout, stderr string) {
+	go proxyIO(ctx, stdin, internal.StdinPort, true)
+	go proxyIO(ctx, stdout, internal.StdoutPort, false)
+	go proxyIO(ctx, stderr, internal.StderrPort, false)
+}
+
+func proxyIO(ctx context.Context, path string, port uint32, in bool) {
+	if path == "" {
+		return
+	}
+	log.G(ctx).Debug("setting up IO for " + path)
+	f, err := fifo.OpenFifo(ctx, path, syscall.O_RDWR|syscall.O_NONBLOCK|syscall.O_CREAT, 0700)
+	if err != nil {
+		log.G(ctx).WithError(err).Error("error opening fifo")
+		return
+	}
+	listener, err := vsock.Listen(port)
```

Comment:
I think this makes sense. We also do need to add tests. I will try to take a pass at both on the next revision.

Reply:
I dislike mocks with great passion. I dislike restructuring code to facilitate mocks even more.

Reply:
@nmeyerhans - I disagree. We need to have tests, and we need some way of doing that and right now we have to mock since there isn't any other options without doing a full blown functional test. The benefit of this, also, is it also allows for extendability.

We shouldn't use the excuse of not liking something to avoid writing tests.

Reply:
I don't think we need separate bridge here. vsock already implements `net.Listener` / `net.Conn`, they are already generic interfaces for both vsock and net tcp/udp things.

Reply:
Yea, except you can't swap it in the `net.Listener` easily. That's the point of the bridge

Reply:
Can be same trick as with ioctl ?

Reply:
@mxpv - Talked with Cody, and he stated he could refactor things later to allow for the `net.Listener` to be passed in which would solve the issue. So, I am okay with adding the test later when we refactor this code.